src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -456,6 +456,10 @@ private static async Task<object> ExecuteHubMethod(ObjectMethodExecutor methodEx
                 {
                     await (Task)methodExecutor.Execute(hub, arguments);
                 }
+                else if (methodExecutor.MethodReturnType == typeof(ValueTask))
+                {
+                    await (ValueTask)methodExecutor.Execute(hub, arguments);
```

Comment:
Does this change just avoid boxing/statemachine from the case at L465?

Reply:
That's the hypothesis